doc/feature/spikeReport.md

```diff
@@ -0,0 +1,476 @@
+         *         post-conditions of read in the receiving end.
+         */
+        void write( const const std::vector< float, uint32_t >& spikes );
+
+        /* Overload of the method above provided for convenience */
+        void write( const Spikes& spikes );
+
+        /*
+         * Return the start time of the report if known, UNDEFINED_TIMESTAMP
+         * otherwise.
+         *
+         * In file based reports it's the time of the first spike or undefined
+         * if the report is empty.
+         * In stream based reports it's always 0.
+         */
+        float getStartTime() const;
+
+        /*
+         * Return the end time of the report if known, UNDEFINED_TIMESTAMP
+         * otherwise.
+         *
+         * In file based reports the end time is the timestamp of the last spike
+         * plus an epsilon, or undefined if the report is empty or it cannot be
+         * be decided without loading the whole data.
+         * In stream based reports it's always undefined until the report is
+         * closed. When the report is closed it will be the latest timestamp
+         * reported by the sender or the time of the last spike + epsilon,
+         * whichever is larger; or 0 if no information was received.
+         *
+         * To guarantee const-correctness the value returned by this function
+         * can only change when read is called.
+         */
+        float getEndTime() const;
+
+        /*
+         * Closes the report.
+         *
+         * Only meaningful for stream based reports.
+         * For reports opened in write mode it finishes the reporting. For
+         * reports opened in read mode it disconnects from the source, any call
+         * waiting in read will be unblocked and getEndTime will return
+         * the last known timestamp.
+         *
+         * This function is thread-safe with read as it ensures that any
+         * thread waiting there with an infinite timeout will eventually return.
+         *
+         * Implicitly called by the destructor, (but calling the destructor and
+         * read at the same time is not safe!). Calling any other function
+         * after the report has been closed has undefined behavior.
+         */
+        void close();
+
+        /*
+         * Clears all internal caching.
+         *
+         * Any Spikes container returned by this class will remain valid as
+         * expected.
+         *
+         * The implementation is allowed to retain as much memory as needed
+         * until all pre-existing Spikes containers are destroyed to guarantee
+         * this.
+         */
+        void clear();
+    }
+
+
+## Examples
+
+### Copying a spike report from one URL to another
+
+    SpikeReport source( "url1", brion::MODE_READ )
+    SpikeReport destination( "url2", brion::MODE_WRITE )
+
+    float last = 0;
+    // In file based reports this will iterates just once.
+    while( source.getEndTime() == brion::UNDEFINED_TIMESTAMP )
```

Comment:
how about having an END_OF_REPORT instead of UNDEFINED. That is not informative at all. It can underneath be undefined but for a user using the API, it is confusing.